src/HashBus.Twitter.Monitor.CatchUp/App.cs

```diff
@@ -1,36 +1,47 @@
 ﻿namespace HashBus.Twitter.Monitor.CatchUp
 {
     using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using HashBus.Application.Commands;
     using HashBus.NServiceBusConfiguration;
+    using HashBus.Twitter.Monitor.CatchUp.Commands;
+    using HashBus.Twitter.Monitor.Events;
     using NServiceBus;
     using NServiceBus.Persistence;
 
     class App
     {
-        public static void Run(
-            int maximumNumberOfTweetsPerCatchUp,
-            TimeSpan defaultTransactionTimeout,
-            string nserviceBusConnectionString,
-            string endpointName,
-            string consumerKey,
-            string consumerSecret,
-            string accessToken,
-            string accessTokenSecret)
+        public static async Task Run(int maximumNumberOfTweetsPerCatchUp, TimeSpan defaultTransactionTimeout, string nserviceBusConnectionString, string endpointName, 
+            string consumerKey, string consumerSecret, string accessToken, string accessTokenSecret, string monitorCatchUpAddress, string applicationAddress, string monitorAddress)
         {
-            var busConfiguration = new BusConfiguration();
-            busConfiguration.Transactions().DoNotWrapHandlersExecutionInATransactionScope();
-            busConfiguration.Transactions().DefaultTimeout(defaultTransactionTimeout);
-            busConfiguration.EndpointName(endpointName);
-            busConfiguration.UseSerialization<JsonSerializer>();
-            busConfiguration.EnableInstallers();
-            busConfiguration.UsePersistence<NHibernatePersistence>().ConnectionString(nserviceBusConnectionString);
-            busConfiguration.ApplyMessageConventions();
-            busConfiguration.RegisterComponents(c=>c.RegisterSingleton<ITweetService>(
+            var endpointConfiguration = new EndpointConfiguration(endpointName);
+            endpointConfiguration.UseTransport<MsmqTransport>().Transactions(TransportTransactionMode.ReceiveOnly);
+            endpointConfiguration.UnitOfWork().WrapHandlersInATransactionScope(defaultTransactionTimeout);
+            endpointConfiguration.UseSerialization<JsonSerializer>();
+            endpointConfiguration.EnableInstallers();
+            endpointConfiguration.UsePersistence<NHibernatePersistence>().ConnectionString(nserviceBusConnectionString);
+            endpointConfiguration.ApplyMessageConventions();
+            endpointConfiguration.ApplyErrorAndAuditQueueSettings();
+            endpointConfiguration.RegisterComponents(c=>c.RegisterSingleton<ITweetService>(
                 new TweetService(maximumNumberOfTweetsPerCatchUp, consumerKey, consumerSecret, accessToken, accessTokenSecret)));
 
-            using (Bus.Create(busConfiguration).Start())
+            var transportExtensions = endpointConfiguration.UseTransport<MsmqTransport>();
+            var routing = transportExtensions.Routing();

```

Comment:
Please insert a blank line before this line (for consistency with other endpoints).